sound/soc/sof/Kconfig

```diff
@@ -87,6 +87,27 @@ config SND_SOC_SOF_STRICT_ABI_CHECKS
 	  If you are not involved in SOF releases and CI development
 	  select "N".
 
+choice
+	prompt "Virtualisation role"
+	default SND_SOC_SOF_VIRTIO_NONE
+	depends on VIRTIO
+	---help---
+	  If this kernel should be used in a virtualised environment, select
+	  which role it will be performing.
+
+config SND_SOC_SOF_VIRTIO_NONE
+	bool "None"
+	---help---
+	  No virtualisation
+
+config SND_SOC_SOF_VIRTIO_FE
+	bool "Guest"
+	depends on SND_SOC_SOF_NOCODEC_SUPPORT
```

Comment:
NOCODEC still means you expose SSP and DMICs DAIs, is this what you want?
I think you probably have a concept of NO DAI, or VIRTUAL_DAI only.

Reply:
@plbossart I think the answer to this your question is - yes, I'm using virtual DAIs. As for the use of NOCODEC - I'm indeed not sure about it. It was there in the original version and it seems to do the job. Whether we really want to keep using it or to switch to something different - now or later - not sure. @lgirdwood ?

Reply:
if we want the same kernel to be used either natively or as a guest, then this dependency has to away since it would not work e.g. for HDaudio platforms.

Reply:
@plbossart I'm not sure how important the use of nocodec by the vFE is. I guess it allows for some code re-use, and duplicating the code wouldn't be good, unless it's just something small and trivial. Can we not fix the nocodec conflict with hdaudio and others? Is there a principal problem there?